### PR TITLE
[PA-5586]: Add debian-12-x86_64 to the puppet agent module task acceptance

### PR DESCRIPTION
**Pull request by span786**
Added debian-12 to the puppet agent module task acceptance

### CODE REVIEW

File: task_spec/spec/acceptance/init_spec.rb [hunk 1]
```diff
@@ -38,6 +38,19 @@ def log_output_errors(result)
     puts logger.info(out)
   end
 
+  # Added this method to simplify the 'case' condition
+  # used for target_platform, which will use latest puppet_agent
+  # in below mentioned test spec
+  def latest_platform_list
+    %r{
+      el-9-aarch64|
+      ubuntu-22.04-aarch64|
```
Comment (kenyon) [inline task_spec/spec/acceptance/init_spec.rb:47]:
```suggestion
      ubuntu-22\.04-aarch64|
```

**Review by kenyon — APPROVED** [review]
Minor thing, since these are regexps, the dot can match any character, which shouldn't cause an unexpected match, but I always like to escape characters that should be literal instead of regexp metacharacters. You could also use Regexp.escape here instead.

File: task_spec/spec/acceptance/init_spec.rb (hunk 1 above)
Comment (span786) [reply task_spec/spec/acceptance/init_spec.rb]: Thank you @kenyon. Good catch.

I have modified the regex. Cheers!!

File: task_spec/spec/acceptance/init_spec.rb [hunk 2]
```diff
@@ -38,6 +38,19 @@ def log_output_errors(result)
...
+  # Added this method to simplify the 'case' condition
+  # used for target_platform, which will use latest puppet_agent
+  # in below mentioned test spec
+  def latest_platform_list
+    %r{
+      el-9-aarch64|
+      ubuntu-22\.04-aarch64|
+      amazon-2023|
+      osx-14|
+      debian-12
```
Comment (skyamgarp) [inline task_spec/spec/acceptance/init_spec.rb:50]: Hello Saurabh,
I think we can remove above other platforms as those are released and versions for them can be updated. But we might need a different jira for same.

**Review by kenyon — APPROVED** [review]

**Review by kenyon — APPROVED** [review]

**Comment by span786** [comment]
@bastelfreak can you please review this PR ?
It is not allowing to merge without code owner approval.

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**